src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/ciagmigration/CiagInductionMigrationService.kt

```diff
@@ -12,22 +12,34 @@ private val log = KotlinLogging.logger {}
  * API and database.
  */
 @Service
-class CiagInductionMigrationService(private val ciagWebClient: WebClient) {
+class CiagInductionMigrationService(
+  private val timelineLookupService: PrisonerInductionTimelineLookupService,
+  private val ciagWebClient: WebClient,
+) {
 
   fun migrateCiagInductions() {
     log.info { "Starting migration of CIAG Inductions from CIAG API to PLP API" }
 
     try {
-      val ciagInduction = ciagWebClient
-        .get()
-        .uri("/ciag/induction/A5077DY")
-        .retrieve()
-        .bodyToMono(CiagInductionResponse::class.java)
-        .block()
-
-      log.info { "Deserialised CIAG Induction: $ciagInduction" }
+      val prisonNumbers = timelineLookupService.getPrisonNumbersToMigrate()
+      log.info { "Identified ${prisonNumbers.size} Inductions to migrate" }
+
+      prisonNumbers.forEach {
+        val ciagInduction = getCiagInduction(it)
+        log.info { "Deserialized Induction for prisoner ${ciagInduction!!.offenderId}" }
```

Comment:
It will be interesting to see if this works. I suspect there's all kind of dodgy data in dev, so I'm half expecting this to fail. If it does then we need to find out why and what we next (e.g. delete the Induction(s) in question, or fix the ccde).